jwt/api_jwt.py

```diff
@@ -412,19 +416,30 @@ def _validate_aud(
         if all(aud not in audience_claims for aud in audience):
             raise InvalidAudienceError("Audience doesn't match")
 
-    def _validate_iss(self, payload: dict[str, Any], issuer: Any) -> None:
+    def _validate_iss(
+        self, payload: dict[str, Any], issuer: list[str] | str | None
```

Comment:
It's usually a good idea to accept wider types in function args (but have narrow type for the return value)
```suggestion
        self, payload: dict[str, Any], issuer: typing.Iterable[str] | str | None
```

Reply:
yep, oversight since I was looking at [this PR](https://github.com/jpadilla/pyjwt/pull/913/files) (which specifies issuer should be list[str], not just a sequence) when I was working on the changes.

Looking at it again, this should be a Container or Sequence given we're doing a `x in y` check. Per your point, picking Container to keep the type wide as possible